### ISSUE TITLE
Automatic refreshing browser when --static-dir is used

### ISSUE THREAD

**Issue by fregiese** [storybookjs/storybook#11406]
### TLDR
Automatic refresh of the browser, when files in **--static-dir** are changed.

### Long Version
In our project we use the **--static-dir** CLI Option to load CSS and JS files from a static folder, as explained here: https://storybook.js.org/docs/configurations/serving-static-files/#2-via-a-directory

The Problem is that only the changes on stories are being refreshed in the browser. To refresh CSS and JS, we have to manually refresh the browser, so the files from the **--static-dir** are updated in storybook.

We need to use **--static-dir**, because we have a complex webpack configuration to create our CSS and JS files, that's why we don't integrate the CSS and JS integration, into the storybook webpack configuration. Our custom webpack config is used with **--watch**, so the CSS and JS are always rebuild, when we change CSS or JS.

Is there a way to auto-refresh the browser, when files in **--static-dir** are changed? There doesn't seem to be a **--watch** option on **start-storybook** https://storybook.js.org/docs/configurations/cli-options/#for-start-storybook only for **build-storybook**.

The only related issue I could find was a closed feature request for HMR: https://github.com/storybookjs/storybook/issues/9992

**Comment by norin89** [comment]
@fregiese I am facing the same issue. Did you manage to solve it somehow?

**Comment by dpxgit** [comment]
@norin89 We have implemented HMR with a custom `webpackDevServer`, but it was a lot of work. It would have been much easier if Storybook would have an option for this. We are starting two processes now, one with storybook and one with our custom webpack config and the `webpackDevServer`. The paths for css and js in our html templates are linking to `localhost:3001` now and would be different for every project setup.


This is the config for our `webpackDevServer`, but the rest is different for every project setup.

webpack.config.js:
```js
devServer: {
  contentBase: outputPath, // outputPath: the path to the dist folder where the css and js is located
  compress: true,
  port: 3001,
  hot: true,
  disableHostCheck: true,
  writeToDisk: true,
},
```
